### DISCUSSION TITLE
Unit testing

### DISCUSSION THREAD

**Discussion by nandanvasudevan** [cycfi/elements#348] [Q&A]
Hi,

I recently tried to unit test an application built using Gtkmm 3.0, I failed even doing the basic setup needed.
I don't have a minimal example, but that can be done.

I just want to know how easy or difficult it would be to unit test widgets or elements created.

Can I test the elements without initializing an app instance?

My testing framework of choice is Catch2.

**Comment by redtide** [comment]
See also #351
